mne/simulation/raw.py

```diff
@@ -314,11 +344,22 @@ def simulate_raw(raw, stc, trans, src, bem, cov='simple',
                 % (len(dev_head_ts), _pl(dev_head_ts), interp))
     del interp
 
-    verts = stc.vertices
-    verts = [verts] if isinstance(stc, VolSourceEstimate) else verts
+    # do first iter so we can get the vertex subselection
+    # Ensure STC is iterable
+    if not isinstance(stc, Iterable) or (
+            isinstance(stc, (tuple, list)) and
+            len(stc) == 2 and isinstance(stc[0], _BaseSourceEstimate) and not
+            isinstance(stc[1], _BaseSourceEstimate)):
+        logger.info('Turning stc input into iterable using cycle')
+        stc = cycle([stc])
+    stc = iter(stc)
+
+    stc_count = 0
+    first_stc = next(stc)
+    verts = _next_stc_event_data(first_stc, 1, stc_count, info['sfreq'])
```

Comment:
`_next_stc_event_data` returns two variables - `stc_data, event_data`, which contain data, not vertices information. But it's saved to `verts` and then it goes as an input to `_restrict_source_space_to(src, verts)`, which expects vertices info, not data.

Reply:
This wasn't actually true -- the output depended on the inputs (sometimes the output was verts, sometimes data/events). But I've changed it not to do that conditional output, now it's always data/events/verts.